lib/src/main/java/graphql/nadel/engine/transform/partition/NadelFieldPartition.kt

```diff
@@ -0,0 +1,167 @@
+package graphql.nadel.engine.transform.partition
+
+import graphql.VisibleForTesting
+import graphql.language.ScalarValue
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.NormalizedInputValue
+import graphql.schema.GraphQLArgument
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLInputObjectType
+import graphql.schema.GraphQLInputValueDefinition
+import graphql.schema.GraphQLSchema
+import graphql.schema.GraphQLTypeUtil
+
+class NadelFieldPartition(
+    private val pathToPartitionArg: List<String>,
+    private val fieldPartitionContext: Any,
+    private val graphQLSchema: GraphQLSchema,
+    private val partitionKeyExtractor: (
+        scalarValue: ScalarValue<*>,
+        inputValueDef: GraphQLInputValueDefinition,
+        context: Any,
+    ) -> String?,
```

Comment:
Would like to see this made into an interface so it's easier to parse e.g.

```kotlin
interface NadelPartitionKeyExtractor {
  fun getPartitionKey(
        scalarValue: ScalarValue<*>,
        inputValueDef: GraphQLInputValueDefinition,
        context: Any,
    ): String
}
```

Reply:
Nice.